shared_task_code/ecs_task_queue_consumer.py

```diff
@@ -0,0 +1,131 @@
+import os
+import aioboto3
+import boto3
+from utils.lambda_decorators import ssm_parameters, async_handler
+from utils.json_serialisation import dumps
+from json import loads
+from datetime import datetime
+import subprocess
+
+
+class ECSTaskQueueConsumer:
+    region = os.environ["REGION"]
+    stage = os.environ["STAGE"]
+    app_name = os.environ["APP_NAME"]
+    task_name = os.environ["TASK_NAME"]
+    ssm_prefix = f"/{app_name}/{stage}"
+    ssm_client = boto3.client("ssm", region_name=region)
+    ecs_client = boto3.client("ecs", region_name=region)
+
+    PRIVATE_SUBNETS = f"{ssm_prefix}/vpc/using_private_subnets"
+    SUBNETS = f"{ssm_prefix}/vpc/subnets/instance"
+    CLUSTER = f"{ssm_prefix}/ecs/cluster"
+    RESULTS = f"{ssm_prefix}/tasks/{task_name}/s3/results/id"
+    SECURITY_GROUP = f"{ssm_prefix}/tasks/{task_name}/security_group/id"
+    IMAGE_ID = f"{ssm_prefix}/tasks/{task_name}/image/id"
+
+    def get_ssm_params(self, ssm_client, param_names):
+        ret = self.ssm_client.get_parameters(Names=param_names)
+        params = ret['Parameters']
+        return {p['Name']: p['Value'] for p in params}
+
+    def __init__(self, event):
+        self.event = event
+
+    def submit_ecs_task(self, event, task_params, message_id):
+        ssm_params = event["ssm_params"]
+        private_subnet = "true" == ssm_params[self.PRIVATE_SUBNETS]
+        network_configuration = {
+            "awsvpcConfiguration": {
+                "subnets": ssm_params[self.SUBNETS].split(","),
+                "securityGroups": [ssm_params[self.SECURITY_GROUP]],
+                "assignPublicIp": "DISABLED" if private_subnet else "ENABLED"
+            }
+        }
+        ecs_params = {
+            "cluster": ssm_params[self.CLUSTER],
+            "networkConfiguration": network_configuration,
+            "taskDefinition": ssm_params[self.IMAGE_ID],
+            "launchType": "FARGATE",
+            "overrides": {
+                "containerOverrides": [{
+                    "name": self.task_name,
+                    "environment": [
+                        {
+                            "name": "TASK_INPUT",

```

Comment:
Every time we talked about this piece of work I suggested that this class define a method that maps the task params to a set of environment variables, not how we here have only a single environment variable.

For a more complex scan that needs multiple parameters we would end up serialising and deserialising json through this one env var which is an unnecessary cost and less convenient to program

We need to subclass this class anyway for each scan because of the validation.

The behaviour here for TASK_INPUT makes sense as a default implementation of this new method

Reply:
for tasks that want to do this, they can create an env_vars array that's passed in to task_params by implementing a validate_data function